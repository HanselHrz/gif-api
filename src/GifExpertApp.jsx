import { useState } from "react";
import { AddCategory, GifGrid } from "./components";
export const GifExpertApp = () => {
  //no poner condicionalmente los hooks
  const [categories, setCategories] = useState(["mr. robot", "rick and morty"]);

  const onAddCategory = (onNewCategory) => {
    if (categories.includes(onNewCategory)) return;
    setCategories([onNewCategory, ...categories]);
    // setCategories([...categories, value]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <hr />
      <AddCategory onNewCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
