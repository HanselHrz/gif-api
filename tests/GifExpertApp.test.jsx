import { render } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("pruebas en el componente < GifExpertApp />", () => {
  test("Debe de hacer match con el snapshot ", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
  });
});
