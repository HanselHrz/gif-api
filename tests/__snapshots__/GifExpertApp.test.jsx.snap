// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente < GifExpertApp /> Debe de hacer match con el snapshot  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar GIF"
      type="text"
      value=""
    />
  </form>
  <h3>
    mr. robot
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    rick and morty
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`pruebas en el componente < GifExpertApp /> should  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar GIF"
      type="text"
      value=""
    />
  </form>
  <h3>
    mr. robot
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    rick and morty
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
